input/tar2/tar_input.go

```diff
@@ -0,0 +1,163 @@
+package tar2
+
+import (
+	"archive/tar"
+	"bytes"
+	"crypto/sha512"
+	"encoding/base64"
+	"hash"
+	"io"
+	"os"
+	"path"
```

Comment:
Sounds good. I'd be comfortable swapping them out once there's something better.